src/test/lib/HandshakeTest.cpp

```diff
@@ -3214,7 +3216,10 @@ QuicTestChangeAlpn(
                                 QuicAddrGetFamily(&ServerLocalAddr.SockAddr)),
                             ServerLocalAddr.GetPort()));
 
-                    Client.SetExpectedTransportCloseStatus(QUIC_STATUS_INTERNAL_ERROR);
+                    // Make sure expected transport close status has been transferred to Server Conn
+                    if (!ServerAcceptCtx.WaitForNewConnection(3000)) {
+                        return;
+                    }
```

Comment:
Why is this needed? I think the other line above is the actual fix, no?

Reply:
No, I saw the unexpected error comes from the server side; it means we weren't able to set `ExpectedTransportCloseStatus`

Reply:
```
[0]0B98.0CAC::2022/08/19-18:32:50.198116500 [Microsoft-Quic][conn][0x12765D8D860] Created, IsServer=1, CorrelationId=11
[0]0B98.18A0::2022/08/19-18:32:50.198354100 [Microsoft-Quic][conn][0x12765D8D860] ERROR, No ALPN match found.
```

Reply:
This shouldn't effect the server side setting the expected transport close status. That happens in the listener callback (see `ListenerAcceptConnection`). Could you reproduce this problem locally?

Reply:
No, that's why I wanted to do it in this way, but I can remove it, and we can try with the above lines only

Reply:
Oh, that's right, the actual fix must be the `Client` line, looks like I mixed up the connection addresses.